Makefile

```
@@ -124,11 +128,23 @@ docker-push-master: 	  ## Push Docker image to registry IF we are currently on t
 		((! (git diff HEAD~1 localstack/__init__.py | grep '^+__version__ =') && \
 			echo "Only pushing tag 'latest' as version has not changed.") || \
 			(docker tag $(IMAGE_NAME):latest-$(PLATFORM) $(IMAGE_NAME):$(IMAGE_TAG)-$(PLATFORM) && \
+				docker tag $(IMAGE_NAME):$(MAJOR_VERSION).$(MINOR_VERSION)-$(PLATFORM) && \
+				docker tag $(IMAGE_NAME):$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)-$(PLATFORM) && \
 				docker tag $(IMAGE_NAME_LIGHT):latest-$(PLATFORM) $(IMAGE_NAME_LIGHT):$(IMAGE_TAG)-$(PLATFORM) && \
+				docker tag $(IMAGE_NAME_LIGHT):$(MAJOR_VERSION).$(MINOR_VERSION)-$(PLATFORM) && \
+				docker tag $(IMAGE_NAME_LIGHT):$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)-$(PLATFORM) && \
 				docker tag $(IMAGE_NAME_FULL):latest-$(PLATFORM) $(IMAGE_NAME_FULL):$(IMAGE_TAG)-$(PLATFORM) && \
+				docker tag $(IMAGE_NAME_FULL):$(MAJOR_VERSION).$(MINOR_VERSION)-$(PLATFORM) && \
+				docker tag $(IMAGE_NAME_FULL):$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)-$(PLATFORM) && \
 				docker push $(IMAGE_NAME):$(IMAGE_TAG)-$(PLATFORM) && \
+				docker push $(IMAGE_NAME):$(MAJOR_VERSION).$(MINOR_VERSION)-$(PLATFORM) && \
+				docker push $(IMAGE_NAME):$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)-$(PLATFORM) && \
 				docker push $(IMAGE_NAME_LIGHT):$(IMAGE_TAG)-$(PLATFORM) && \
+				docker push $(IMAGE_NAME_LIGHT):$(MAJOR_VERSION).$(MINOR_VERSION)-$(PLATFORM) && \
+				docker push $(IMAGE_NAME_LIGHT):$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)-$(PLATFORM) && \
 				docker push $(IMAGE_NAME_FULL):$(IMAGE_TAG)-$(PLATFORM))) && \
+				docker push $(IMAGE_NAME_FULL):$(MAJOR_VERSION).$(MINOR_VERSION)-$(PLATFORM) && \
+				docker push $(IMAGE_NAME_FULL):$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)-$(PLATFORM) && \
```

There's an issue with the brackets here. You actually push the `<major>.<minor>` and the `<major>.<minor>.<patch>` even if the version hasn't changed.
```suggestion
				docker push $(IMAGE_NAME_FULL):$(IMAGE_TAG)-$(PLATFORM) && \
				docker push $(IMAGE_NAME_FULL):$(MAJOR_VERSION).$(MINOR_VERSION)-$(PLATFORM) && \
				docker push $(IMAGE_NAME_FULL):$(MAJOR_VERSION).$(MINOR_VERSION).$(PATCH_VERSION)-$(PLATFORM))) && \
```